engine/classes/ElggPlugin.php

```diff
@@ -814,7 +814,19 @@ protected function canReadFile($filename) {
 	 * @return void
 	 */
 	protected function registerViews() {
-		if (!_elgg_services()->views->registerPluginViews($this->path, $failed_dir)) {
+		$views = _elgg_services()->views;
+
+		// Declared views first
+		$file = "{$this->path}/views.php";
+		if (is_file($file)) {
+			$spec = (include $file);
+			if (is_array($spec)) {
+				$views->mergeViewsSpec($spec);
+			}
+		}
+
+		// Allow /views directory files to override
```

Comment:
@hypeJunction views.php is registered first, then the views/ tree is walked so any views in there will override.